From e1c2ce7f0edb092c4d735a0c214a33e77a8dcd13 Mon Sep 17 00:00:00 2001
From: Justin Kromlinger <mail@hashworks.net>
Date: Tue, 4 Aug 2026 16:50:03 +0200
Subject: [PATCH] Fix build with ICU 76

Due to unicode-org/icu@199bc82, ICU 76 no longer adds `icu-uc` by
default. This causes linker errors for undefined symbols.

Meson will automatically flatten the dependencies list as documented
at https://mesonbuild.com/Reference-manual_functions.html#build_target

See https://github.com/openzim/libzim/blob/9.8.1/meson.build#L84
---
 meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/meson.build b/meson.build
index ee307d2..7fb3e78 100644
--- a/meson.build
+++ b/meson.build
@@ -26,6 +26,9 @@ find_library_in_compiler = meson.version().version_compare('>=0.31.0')
 rt_dep = dependency('rt', required:false)
 docopt_dep = dependency('docopt', static:static_linkage)
 icu_dep = dependency('icu-i18n', static:static_linkage)
+if icu_dep.version().version_compare('>= 76')
+  icu_dep = [icu_dep, dependency('icu-uc', static:static_linkage)]
+endif
 
 with_writer = host_machine.system() != 'windows'
 
-- 
2.55.0

